tap_github/streams.py

```diff
@@ -412,3 +423,254 @@ def post_process(self, row: dict, context: Optional[dict] = None) -> dict:
             ),
         ),
     ).to_dict()
+
+
+class PullRequestsStream(GitHubStream):
+    """Defines 'PullRequests' stream."""
+
+    name = "pull_requests"
+    path = "/repos/{org}/{repo}/pulls"
+    primary_keys = ["id"]
+    replication_key = "updated_at"
+    parent_stream_type = RepositoryStream
+    ignore_parent_replication_key = False
+    state_partitioning_keys = ["repo", "org"]
+
+    def get_url_params(
+        self, context: Optional[dict], next_page_token: Optional[Any]
+    ) -> Dict[str, Any]:
+        """Return a dictionary of values to be used in URL parameterization."""
+        assert context is not None, f"Context cannot be empty for '{self.name}' stream."
+        params = super().get_url_params(context, next_page_token)
+        # Fetch all pull requests regardless of state (OPEN, CLOSED, MERGED).
+        params["state"] = "all"
+        return params
+
+    def get_child_context(self, record: dict, context: Optional[dict]) -> dict:
+        """Return a child context object from the record and optional provided context.
+
+        By default, will return context if provided and otherwise the record dict.
+        Developers may override this behavior to send specific information to child
+        streams for context.
+        """
+        if context is None:
+            raise ValueError("Issue stream should not have blank context.")
+
+        context["issue_number"] = record["number"]
+        context["comments"] = record["comments"]  # If zero, comments can be skipped
+        return context
```

Comment:
Since PullRequests don't have child stream types any more, I think this method is not used and can be safely removed. (Probably also not used for `IssuesStream` anymore but that's a separate topic.)